docs/admin_api/rooms.md

```diff
@@ -475,16 +524,82 @@ The following JSON body parameters are available:
 
 The JSON body must not be empty. The body must be at least `{}`.
 
+## Status of deleting rooms
+
+**Note**: This API is new, experimental and "subject to change".
+
+It is possible to query the status of the background task for deleting rooms.
+The status can be queried up to 24 hours after completion of the task,
+or until Synapse is restarted (whichever happens first).
+
+With this API you can get the status of all tasks the last 24h for this `room_id`.
+If you want only get the status of one specific task, you can also use
+the [Purge status query API](purge_history_api.md#Purge-status-query).
+
+The API is:
+
+```
+GET /_synapse/admin/v2/rooms/<room_id>/delete_status
+```
+
+A response body like the following is returned:
+
+```json
+{
+    "delete_status": [
+        {
+            "purge_id": "purgeid1",
+            "status": "failed",
+            "result": {
+                "kicked_users": [],
+                "failed_to_kick_users": [],
+                "local_aliases": [],
+                "new_room_id": null
+            }
+        }, {
+            "purge_id": "purgeid2",
+            "status": "active",
+            "result": {
+                "kicked_users": [
+                    "@foobar:example.com"
+                ],
+                "failed_to_kick_users": [],
+                "local_aliases": [
+                    "#badroom:example.com",
+                    "#evilsaloon:example.com"
+                ],
+                "new_room_id": "!newroomid:example.com"
+            }
+        }
+    ]
+}
+```
+
+**Parameters**
+
+The following parameters should be set in the URL:
+
+* `room_id` - The ID of the room.
+
 **Response**
 
 The following fields are returned in the JSON response body:
 
-* `kicked_users` - An array of users (`user_id`) that were kicked.
-* `failed_to_kick_users` - An array of users (`user_id`) that that were not kicked.
-* `local_aliases` - An array of strings representing the local aliases that were migrated from
-                    the old room to the new.
-* `new_room_id` - A string representing the room ID of the new room.
-
+- `delete_status` - An array of objects, each containing information about one task.
+  Task objects contain the following fields:
+  - `status` - The status will be one of:
+    - `remove members` - The process is removing users from the `room_id`.
+    - `active` - The process is purging the room from database.
```

Comment:
`active` is confusing here. How about:

```suggestion
    - `shutting_down` - The process is removing users from the room.
    - `purging` - The process is purging the room and event data from database.
```

Reply:
If I change `active` to `purging` I change the `/purge_history_status` API.
If this is okay. I do this. Otherwise I have to create a new status.

Reply:
Sytest fails.
https://github.com/matrix-org/sytest/blob/57c856c0c03f41ef9a6d06c89ca898e594505a4b/tests/48admin.pl#L28

Should I create an patch for sytest?
`   }, while => sub { $_[0]->get eq 'active' || $_[0]->get eq 'purging' });`

Reply:
And we certainly need an upgrade notice.

Reply:
Sorry, I think this needs to be specific to delete requests.

I think this, and the problem with `asdict`, are really symptoms of re-using `PurgeStatus` for a different type of request.

Reply:
Ok. I will create a new class `DeleteStatus` to solve the problem.

Reply:
Thanks. Sorry this is turning into a bit of an epic!